src/main/java/org/aksw/simba/lemming/metrics/single/edgemanipulation/EdgeModifier.java

```diff
@@ -27,6 +33,8 @@
 	private List<TripleBaseSingleID> mLstAddedEdges;
 	private boolean isCoutingEdgeTriangles = false;
 	private boolean isCountingNodeTriangles = false;
+	private HashMap<String, UpdatableMetricResult> mMapPrevMetricsResult;
+	private List<SingleValueMetric> metricsOptimized;
```

Comment:
Why is this separation between "optimized" metrics and the other list of metrics necessary?

Reply:
For metrics present in the metricsOptimized list, the applyUpdatable method is invoked.  This method saves the result object of different metrics computed by the update method. This result object can be for remove an edge or add an edge depending on the error score. To ensure that the applyUpdatable method is invoked for the metrics which are optimized, we are using metricsOptimized list. 
This rough execution flow is also illustrated in the attached diagram.
[Execution flow.pdf](https://github.com/dice-group/Lemming/files/6642306/Execution.flow.pdf)

Reply:
Please note that the API of the metrics is created in a way that you _do not have to distinguish_ between the "optimized" metrics and the "normal" metrics. Hence, the list is not necessary.